File: frontend/src/services/deficit.ts
// src/services/deficit.ts
import api from "@/lib/axios";
import { DeficitMonthlyResponse } from "@/models/deficit";

export const deficitService = {
  async getMonthlyDeficit(year: number, month: number): Promise<DeficitMonthlyResponse> {
    const { data } = await api.get<DeficitMonthlyResponse>("/deficit/monthly", {
      params: { year, month },
    });
    return data;
  },
};

File: backend/app/routers/deficit.py
# app/routers/deficit.py
from fastapi import APIRouter, Depends, Query
from sqlalchemy.orm import Session

from app.dependencies.db import get_db
from app.schemas.deficit import DeficitMonthlyResponse
from app.services.deficit import DeficitService

router = APIRouter(tags=["deficit"])


@router.get("/deficit/monthly", response_model=DeficitMonthlyResponse)
def get_monthly_deficit(
    year: int = Query(..., description="Year (e.g. 2026)"),
    month: int = Query(..., description="Month (1-12)"),
    db: Session = Depends(get_db),
):
    return DeficitService.calculate_monthly_deficit(db, year, month)
